BigInteger support

kartik-gupta-ij opened the pull request:
#6
Fixed using [json-bigint](https://www.npmjs.com/package/json-bigint)

kartik-gupta-ij committed:
Chore: Test Fixed

kartik-gupta-ij committed:
Chore: Big int Support added

kartik-gupta-ij committed:
Chore : packages Updated fixed

generall commented on package.json:
```diff
@@ -98,6 +99,15 @@
     "singleQuote": true,
     "semi": false
   },
+  "pnpm": {
+    "overrides": {
+      "@babel/traverse": "7.23.2",
+      "semver": "7.5.2",
+      "tough-cookie": "4.1.3",
+      "word-wrap": "1.2.4",
+      "graphql": " 16.8.1 "
+    }
+  },
```
why this is changed?

kartik-gupta-ij replied:
Those packages has vulnerabilities

Rendez replied:
I would encourage you to explore more modern alternatives. This one was last published 4 years ago, and it seems to have quite a few [issues opened](https://github.com/sidorares/json-bigint/issues).

I found one that might be a better candidate: https://github.com/Ivan-Korolenko/json-with-bigint/blob/main/json-with-bigint.js (much, much smaller and simpler and perhaps even faster?) from this S.O. discussion: https://stackoverflow.com/a/69644630

kartik-gupta-ij replied:
@Rendez, The vulnerabilities exist in `jest`, `msw`, `@typescript-eslint`, `jest-environment-jsdom`, and `@changesets/cli`, and they have not been resolved yet.

Rendez replied:
It's not just the vulnerabilities, you're adding a massive outdated and unmaintained library, that is my concern.

Rendez replied:
> @Rendez, The vulnerabilities exist in `jest`, `msw`, `@typescript-eslint`, `jest-environment-jsdom`, and `@changesets/cli`, and they have not been resolved yet.

Can't we update these libraries to latest? Do they still have these vulnerabilites after updating?

kartik-gupta-ij replied:
Yes, they have vulnerabilities in the latest version too.

generall commented:
shouldn't this be also changed https://github.com/qdrant/openapi-typescript-fetch/blob/f33b84870048e75efe68e1ebaf6876bf87548ab1/src/fetcher.ts#L97C2-L97C79 ?

kartik-gupta-ij committed:
Chore : JSONbig.stringify(payload) added

kartik-gupta-ij commented:
> shouldn't this be also changed https://github.com/qdrant/openapi-typescript-fetch/blob/f33b84870048e75efe68e1ebaf6876bf87548ab1/src/fetcher.ts#L97C2-L97C79 ?

Fixed

Rendez commented on src/fetcher.ts:
```diff
@@ -14,6 +14,7 @@ import {
   TypedFetch,
 } from './types.js'
+import JSONbig from 'json-bigint'
 
```
I think we should default to using BigInt, as [all modern browsers have now native support for it](https://caniuse.com/bigint). That way we tell the lib to internally avoid using `bignumber.js`:

`const JSONbigNative = JSONbig({ useNativeBigInt: true });`

kartik-gupta-ij committed:
Update dependencies and fix JSON parsing

kartik-gupta-ij committed:
chore : fomating

Rendez commented on tsconfig.base.json:
```diff
@@ -5,7 +5,7 @@
         "esModuleInterop": true,
         "forceConsistentCasingInFileNames": true,
-        "lib": ["dom", "dom.iterable", "es2019"],
+        "lib": ["dom", "dom.iterable", "es2019", "esnext"],
```
Please follow this convention: https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping

Rendez commented on src/jsonParser.ts:
```diff
@@ -0,0 +1,41 @@
+export type Json =
```
Why did you add:
```json
{
 "dependencies": {
    "json-with-bigint": "^2.1.0"
  }
}
```

And then copied the [source](https://github.com/Ivan-Korolenko/json-with-bigint/blob/main/json-with-bigint.js) here?

Rendez replied:
I much rather have the dependency in case new versions with fixes happen, etc.

Rendez commented on test/fetch.test.ts:
```diff
@@ -112,14 +112,14 @@ describe('fetch', () => {
   })
 
   it(`POST /accepted`, async () => {
```
We should test the conversion and parsing of responses with big int.

kartik-gupta-ij committed:
Remove json-with-bigint dependency

kartik-gupta-ij committed:
Update target and module settings in tsconfig files

kartik-gupta-ij committed:
Add bigInt in fetch.test.ts and paths.ts

kartik-gupta-ij committed:
Add Jest configuration and update package dependencies

Rendez committed:
Upgrade workflows node versions

Rendez committed:
Linting and smaller fixes
